### PR TITLE
Switch between emacs and vi keybindings, switch key detection to the keypress module.

### PR DESCRIPTION
**Pull request by ryjm**
This is a bit of a big change. I wanted to have emacs keybindings but it was really hard to get it working with the javascript keycodes. So this requires the keypress module, which doesn't change the current way keys are detected too much, but does make it a bit easier to read (instead of using key codes). You can switch keybindings with the emacs and vi commands. 

Hopefully some of these changes are useful! I'd be happy to change anything if needed.

### CODE REVIEW

**Comment by arscan** [comment]
I'm definitely fine with this change in theory, but I can't seem to get it to run.  I keep getting this:

```
› ./htxt 

/home/rob/Projects/habit-test/index.js:161
});
 ^
SyntaxError: Unexpected token )
    at Module._compile (module.js:437:25)
    at Object.Module._extensions..js (module.js:467:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:362:17)
    at require (module.js:378:17)
    at Object.<anonymous> (/home/rob/Projects/habit-test/htxt:3:1)
    at Module._compile (module.js:449:26)
    at Object.Module._extensions..js (modu
```

Whats strange is that index.js is only 159 lines long... so i guess it has something to do with how index.js is being called?  Can you replicate on your side? Any thoughts?

**Comment by ryjm** [comment]
Something similar happened to me, I'm pretty sure I fixed it by
reinstalling everything with npm. I'll try to replicate later if that
doesn't work.

**Comment by arscan** [comment]
Hmmm, deleting the node_modules directory and doing another 'npm install' didn't seem to help.

**Comment by ryjm** [comment]
Weird. I'll take a look at it tomorrow.

**Comment by ryjm** [comment]
It was just missing a closing brace. oops.

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "superagent": "*",
     "nconf": "*",
     "underscore": "*",
-    "charm": "*"
+    "charm": "*",
+    "keymaster": "*"
```
Comment (arscan) [inline package.json]: should this be keypress?

**Comment by arscan** [comment]
In order to quit in vi mode, I should be able to type ":q", but it gets stuck after the ":" (same with anything in command mode... like :t task to add a task... also the same for search mode).  Are you having the same problems?

**Comment by ryjm** [comment]
Okay I fixed everything (I don't know what was with all the dumb errors, I've got to be more careful when cherry picking from my main branch) and rewrote to a single commit.

On that note: is it preferable to rewrite commits into a single commit for these pull requests, or should I keep the full commit history intact?

**Comment by arscan** [comment]
No worries... I really should have a test suite.

And yeah, I'm working fast and loose on this one at the moment, so feel free to leave the history if that's more convenient for you.

One more thing: could you add the keybindings (defaulted to vi) in config.json.sample?

**Comment by arscan** [comment]
n/m i added the thing to the sample config.

i played around with this -- real cool.  great job.

**Pull request merged**